ui/src/components/FileUpload.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <uppy-dashboard v-if="uppy" :uppy="uppy" :props="uppyDashboardOptions" />
+  <loading-block v-else />
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator'
+import AwsS3Multipart from '@uppy/aws-s3-multipart'
+import Uppy, { UppyFile } from '@uppy/core'
+import { Dashboard as UppyDashboard } from '@uppy/vue'
+import { prepareHeaders } from '@/api'
+import LoadingBlock from './LoadingBlock.vue'
+
+import '@uppy/core/dist/style.css'
+import '@uppy/dashboard/dist/style.css'
+
+const apiURL = window.APP_CONFIG.apiCoreBase
+// Maybe provide this URL in the Hydra API at some point
+const uploadURL = `${apiURL}upload`
+
+export interface UploadedFile extends UppyFile {
+  uploadURL: string
+  s3Multipart: {
+    key: string
+  }
+}
+
+@Component({
+  components: { LoadingBlock, UppyDashboard },
+})
+export default class extends Vue {
+  @Prop() afterUpload: any
+  @Prop() fileMeta: any
+  @Prop({ default: true }) allowMultiple!: boolean
+
+  uppy: Uppy.Uppy<Uppy.StrictTypes> | null = null
+  uppyDashboardOptions = {
+    proudlyDisplayPoweredByUppy: false,
+    showLinkToFileUploadResult: false,
+    note: 'CSV files only',
+    doneButtonHandler: this.onDone,
+  }
+
+  mounted (): void {
+    const uppy = Uppy<Uppy.StrictTypes>({
+      restrictions: {
+        allowedFileTypes: ['.csv'],
+        maxNumberOfFiles: this.allowMultiple ? null : 1,
+      },
+      meta: this.fileMeta,
+    })
+
+    uppy.use(AwsS3Multipart, { companionUrl: uploadURL })
+    uppy.addPreProcessor(async () => {
+      // Hack to set fresh auth token before each upload
+      const headers = prepareHeaders(uploadURL, this.$store)
+      const headersObj = Object.fromEntries(headers.entries());
```

Comment:
You can simplify a little by returning `Record<string, string>` from `prepareHeaders`